crates/syntax/src/algo.rs

```diff
@@ -342,10 +342,10 @@ enum InsertPos {
 
 #[derive(Default)]
 pub struct SyntaxRewriter<'a> {
-    f: Option<Box<dyn Fn(&SyntaxElement) -> Option<SyntaxElement> + 'a>>,
     //FIXME: add debug_assertions that all elements are in fact from the same file.
     replacements: FxHashMap<SyntaxElement, Replacement>,
     insertions: IndexMap<InsertPos, Vec<SyntaxElement>>,
+    _pd: std::marker::PhantomData<&'a ()>,
```

Comment:
This is only here so that the lifetimes dont have to removed, otherwise I will have to rebase #8591 and I'd rather not

Reply:
Canonical name for random PhatomData is :ghost: `ghost` :ghost: